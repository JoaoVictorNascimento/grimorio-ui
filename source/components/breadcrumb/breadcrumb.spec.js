import '../../../internals/test/helper';

import Breadcrumb from './breadcrumb-component';

/** @test {Breadcrumb} */
describe('Breadcrumb component', () => {
  const emptyProps = {
    path: undefined,
    home: undefined,
    onClick: undefined,
    className: undefined,
    onItemClick: undefined,
  }

  let wrapper
  beforeAll(() => {
    wrapper = shallow(
      <Breadcrumb {...emptyProps} />
    );
  })

  describe('#render', () => {
    it('should contains className props', () => {
      const className = 'classeTeste'
      wrapper.setProps({ ...emptyProps, className })
      expect(wrapper.find(`.${className}`)).toHaveLength(1);
    })

    it('when list length is bigger than one', () => {
      const path = 'department/my-amazing-category'
      wrapper.setProps({ ...emptyProps, home: false, path })
      expect(wrapper.debug()).toMatchSnapshot();
    })

    it('when list length is equal one', () => {
      const path = 'department'
      wrapper.setProps({ ...emptyProps, home: false, path })
      expect(wrapper.debug()).toMatchSnapshot();
    })
  });

  describe('call generatePageList', () => {
    it('return list when is NOT home', () => {
      const path = 'department/my-amazing-category'
      wrapper.setProps({ ...emptyProps, home: false, path })
      const resp = wrapper.instance().generatePageList()
      const list = [
        {
          "name": "department",
          "path": "/department",
        },
        {
          "name": "my amazing category",
          "path": "/department/my-amazing-category",
        },
      ]
      expect(resp).toEqual(list);
    });
    it('return list when is home', () => {
      const path = 'department/my-amazing-category'
      wrapper.setProps({ ...emptyProps, home: true, path })
      const resp = wrapper.instance().generatePageList()
      const list = [
        {
          name: true,
          path: "/"
        },
        {
          name: "department",
          path: "/department",
        },
        {
          name: "my amazing category",
          path: "/department/my-amazing-category",
        },
    ]
      expect(resp).toEqual(list);
    })
  })

  it('when click on item', () => {
    const path = 'department/my-amazing-category'
    const props = {
      onItemClick: jest.fn()
    }
    const spy = jest.spyOn(props, 'onItemClick')
    wrapper.setProps({ ...emptyProps, home: false, path, onItemClick: props.onItemClick })

    wrapper.find('span').at(0).props().onClick()
    expect(spy).toHaveBeenCalledWith('/department');
  })
});
