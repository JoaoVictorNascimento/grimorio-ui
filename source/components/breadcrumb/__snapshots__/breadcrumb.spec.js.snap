// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb component #render when list length is bigger than one 1`] = `"<div class=\\"list\\" path=\\"department/my-amazing-category\\"><div class=\\"itemWrap\\"><i style=\\"vertical-align:unset;font-size:16px\\" class=\\"material-icons icon iconHome\\" align=\\"unset\\">home</i><span class=\\"item\\">department</span><i style=\\"vertical-align:middle;font-size:17px\\" class=\\"material-icons icon icon\\" align=\\"middle\\">keyboard_arrow_right</i></div><div class=\\"itemCurrent\\"><span>my amazing category</span></div></div>"`;

exports[`Breadcrumb component #render when list length is equal one 1`] = `"<div class=\\"list\\" path=\\"department\\"><div class=\\"itemWrap\\"><i style=\\"vertical-align:unset;font-size:16px\\" class=\\"material-icons icon iconHome\\" align=\\"unset\\">home</i><span class=\\"itemCurrent\\">department</span></div></div>"`;
